File: proxy/tests/script_engine/url_username.js
/// <reference path="../../script_libs/js/roxy.d.ts" />
/** @type {Extension} */
const url_username = {
  request(flow) {
    if (flow.request.url.username == "dave") {
      flow.request.url.username = "damo";
    }
  },
};
globalThis.extensions = [url_username];

File: proxy/tests/script_engine/url_username.lua
pcall(require, "../../script_libs/lua/roxy.lua")
---@type Extension
local url_username = {
	request = function(flow)
		if flow.request.url.username == "dave" then
			flow.request.url.username = "damo"
		end
	end,
}
Extensions = { url_username }
